pkg/sources/travisci/travisci.go

```diff
@@ -0,0 +1,222 @@
+package travisci
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/go-errors/errors"
+	"github.com/shuheiktgw/go-travis"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/context"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/source_metadatapb"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/sourcespb"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/sources"
+)
+
+const (
+	SourceType = sourcespb.SourceType_SOURCE_TYPE_TRAVISCI
+
+	baseURL = "https://api.travis-ci.com/"
+
+	pageSize = 100
+)
+
+type Source struct {
+	name     string
+	sourceId sources.SourceID
+	jobId    sources.JobID
+	verify   bool
+	jobPool  *errgroup.Group
+	sources.Progress
+	client *travis.Client
+	sources.CommonSourceUnitUnmarshaller
+	returnAfterFirstChunk bool
+}
+
+// Ensure the Source satisfies the interfaces at compile time.
+var _ sources.Source = (*Source)(nil)
+var _ sources.SourceUnitUnmarshaller = (*Source)(nil)
+
+// Type returns the type of source.
+// It is used for matching source types in configuration and job input.
+func (s *Source) Type() sourcespb.SourceType {
+	return SourceType
+}
+
+func (s *Source) SourceID() sources.SourceID {
+	return s.sourceId
+}
+
+func (s *Source) JobID() sources.JobID {
+	return s.jobId
+}
+
+// Init returns an initialized TravisCI source.
+func (s *Source) Init(ctx context.Context, name string, jobId sources.JobID, sourceId sources.SourceID, verify bool, connection *anypb.Any, concurrency int) error {
+	s.name = name
+	s.sourceId = sourceId
+	s.jobId = jobId
+	s.verify = verify
+	s.jobPool = &errgroup.Group{}
+	s.jobPool.SetLimit(concurrency)
+
+	var conn sourcespb.TravisCI
+	if err := anypb.UnmarshalTo(connection, &conn, proto.UnmarshalOptions{}); err != nil {
+		return errors.WrapPrefix(err, "error unmarshalling connection", 0)
+	}
+
+	switch conn.Credential.(type) {
+	case *sourcespb.TravisCI_Token:
+		if conn.GetToken() == "" {
+			return errors.New("token is empty")
+		}
+		s.client = travis.NewClient(baseURL, conn.GetToken())
+		s.client.HTTPClient = common.RetryableHttpClientTimeout(3)
+
+		user, _, err := s.client.User.Current(ctx, nil)
+		if err != nil {
+			return errors.WrapPrefix(err, "error getting testing travis client", 0)
+		}
+		ctx.Logger().V(2).Info("authenticated to Travis CI with user", "username", user.Login)
+	}
+
+	return nil
+}
+
+func (s *Source) Enumerate(ctx context.Context, reporter sources.UnitReporter) error {
+	for repoPage := 0; ; repoPage++ {
+		repositories, _, err := s.client.Repositories.List(ctx, &travis.RepositoriesOption{
+			Limit:  pageSize,
+			Offset: repoPage * pageSize,
+		})
+		if err != nil {
+			if repoPage == 0 {
+				return fmt.Errorf("error listing repositories: %w", err)
+			} else {
```

Comment:
nit: Since the `if` returns we can remove the else and un-indent the following check without needing the else.